### PR TITLE
wip: Use imagemanifest repo instead of the osbuild-composer repository

### PR DESCRIPTION
**Pull request by guy9050**
This PR is dependent on https://github.com/osbuild/automotive-ci/pull/54 .

### CODE REVIEW

File: tests/ci/setup.sh [hunk 1]
```diff
@@ -61,8 +61,21 @@ if [[ "${ID}-${VERSION_ID}" == "centos-8" ]]; then
     cp -fv /etc/redhat-release "${TMPCI_DIR}"
     cp -fv tests/ci/files/rhel-8-4-0-os-release /etc/os-release
     cp -fv tests/ci/files/rhel-8-4-0-rh-release /etc/redhat-release
-    cp -fv /usr/share/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/
-    ln -sfv /etc/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/rhel-8.json
+    #cp -fv /usr/share/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/
+    #ln -sfv /etc/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/rhel-8.json
+
+    #TODO: Make sure REPOSITORY_PATH is initialized
+    echo "{
```
Comment (thrix) [inline tests/ci/setup.sh:70]: @guy9050 please make the template a file

File: tests/ci/setup.sh [hunk 2]
```diff
@@ -61,8 +61,21 @@ if [[ "${ID}-${VERSION_ID}" == "centos-8" ]]; then
     cp -fv /etc/redhat-release "${TMPCI_DIR}"
     cp -fv tests/ci/files/rhel-8-4-0-os-release /etc/os-release
     cp -fv tests/ci/files/rhel-8-4-0-rh-release /etc/redhat-release
-    cp -fv /usr/share/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/
-    ln -sfv /etc/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/rhel-8.json
+    #cp -fv /usr/share/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/
+    #ln -sfv /etc/osbuild-composer/repositories/centos-stream-8.json /etc/osbuild-composer/repositories/rhel-8.json
```
Comment (thrix) [inline tests/ci/setup.sh:69]: I do believe this is still needed as a workaround ...

Comment (juanje) [reply tests/ci/setup.sh:69]: We do. I think the point of creating the yum-repo was for adding it as a new sources (as we already do with the neptune copr), not to use it instead of the default repos.
The adding of the Neptne repo: https://github.com/osbuild/automotive-ci/blob/main/tests/ci/create-commit.sh#L111
The template for the soure: https://github.com/osbuild/automotive-ci/blob/main/tests/ci/files/copr_neptune_tmpl.toml

Mostly, because we aren't going to mirror the whole repository, just some packages. I think. Is it that right,
@lioramilbaum?

The packages from the local yum-repo should be newer than the ones at the official repo, so by just adding the local yum-repo as a new source, `osbuild` should use those packages.
Unless the idea is to mirror the whole CentOS Stream 8 repo, which I think is not.

File: tests/ci/setup.sh [hunk 3]
```diff
@@ -61,8 +61,21 @@ if [[ "${ID}-${VERSION_ID}" == "centos-8" ]]; then
...
+
+    #TODO: Make sure REPOSITORY_PATH is initialized
```
Comment (juanje) [inline tests/ci/setup.sh]: Please, initialize the variable so the script doesn't fail and it doesn't break the rest of the pipeline. Even if that part doesn't work, but at least the scripts try to do the logic and fail there.
You could just define one fake path or do like with the other variables we use from the environment:
```
REPOSITORY_PATH=${REPOSITORY_PATH:-}
```
That won't be assigning any value, but at least it won't fail due `unbound variable`.

File: tests/ci/setup.sh (hunk 2 above)
Comment (thrix) [reply tests/ci/setup.sh:69]: @juanje yeah, this has changed. We are trying to provide the whole repo ourselves from the manifest file. @lioramilbaum is working on generating that repo, and the code @guy9050 is adding here will replace the repos completely. Will see what obstacles we hit with that.

Comment (juanje) [reply tests/ci/setup.sh:69]: I see. That's not what I understood yesterday, so I was confused there.

Then makes sense to add the file as base source, but you're right, the links are needed. We're tricking the `osbuild` to think that the host is RHEL 8.4, instead of CentOS, because the ostree images (the `rhel-edge-commit`and such) are not supported (yet) for CentOS. But then, we need the other trick, to tell the `osbuild` to use the CentOS Stream 8 repos, so it used them and build a (kind of) CentOS Stream 8 system.

TLDR: we need the link or similar solution.

But actually, we might do something like this:
```
envsubst $REPOSITORY_PATH < tests/ci/files/centos-stream-8.json.tmpl > /etc/osbuild-composer/repositories/rhel-8.json
```
And put the content as a template at `tests/ci/files/centos-stream-8.json.tmpl` or similar.

The key thing is to have a file called `/etc/osbuild-composer/repositories/rhel-8.json` with the sources we want the `osbuild` to use.

File: tests/ci/setup.sh [hunk 4]
```diff
@@ -61,8 +61,21 @@ if [[ "${ID}-${VERSION_ID}" == "centos-8" ]]; then
...
+    echo "{
+            "aarch64": [
+              {
+                "name": "automotive",
+                "baseurl": $REPOSITORY_PATH,
+                "enabled": 1,
+                "gpgcheck": 0,
+              },
+            ],
+          }" | envsubst $REPOSITORY_PATH > /usr/share/osbuild-composer/repositories/centos-stream-8.json
```
Comment (juanje) [inline tests/ci/setup.sh]: Please, if you want to change the `centos-stream-8.json` do it so at the `/etc/osbuild-composer/repositories/`, that's the place for files to override the ones from the packages.

File: tests/ci/setup.sh (hunk 1 above)
Comment (guy9050) [reply tests/ci/setup.sh:70]: @thrix do you mean to write the template into a file and then read it, or to pre-create that file?

File: tests/ci/setup.sh (hunk 2 above)
Comment (guy9050) [reply tests/ci/setup.sh:69]: so is there any workaround needed here?

File: tests/ci/setup.sh (hunk 1 above)
Comment (juanje) [reply tests/ci/setup.sh:70]: @guy9050 Yes, it was what we do for the other files (blueprint, ks, etc). Actually, I explain this at the comment above (about the workaround), check after the `TLDR:`.

File: tests/ci/setup.sh (hunk 2 above)
Comment (juanje) [reply tests/ci/setup.sh:69]: Sorry, I wrote a few comments, but I didn't know that they were in some kind of draft :thinking: :sweat_smile: 
Be aware that now we have a temporary fix for the errors with the mirrors:
https://github.com/osbuild/automotive-ci/blob/main/tests/ci/setup.sh#L66

File: tests/ci/setup.sh (hunk 4 above)
Comment (guy9050) [reply tests/ci/setup.sh]: Thanks, fixed

**Comment by lmilbaum** [comment]
The following PR #120 is a duplication of this one. We might want to align the efforts.

**Comment by guy9050** [comment]
Dropping in favor of https://github.com/osbuild/automotive-ci/pull/120
